nio/client/async_client.py

```diff
@@ -917,6 +918,29 @@ def add_response_callback(
 
         return await self._send(KeysQueryResponse, method, path, data)
 
+    @logged_in
+    async def devices(self):
+        # type: () -> Union[DevicesResponse, DevicesError]
```

Comment:
Modern-style type hints (e.g. `devices(self) -> Union[DeviceResponse, DevicesError]`) would be preferred here, the `AsyncClient` is python3-only. Same for `delete_devices`.

Reply:
So modern means adding self? I haven't seen that anywhere else in the code

Reply:
Some methods have them currently:
https://github.com/poljar/matrix-nio/blob/master/nio/client/async_client.py#L1030
https://github.com/poljar/matrix-nio/blob/master/nio/client/async_client.py#L1299
https://github.com/poljar/matrix-nio/blob/master/nio/client/async_client.py#L1403

New functions written for modules that aren't imported in Python 2 (until we drop support for it soon) should follow the new style, which is native Python syntax as opposed to comments.

Reply:
Ah I see, thanks.